<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>异常处理-try/catch</title>
  </head>
  <body>
    <h1>异常处理-try/catch</h1>
    <p>123</p>
    
    <script>
      function fn(){
        //可能发生错误的代码
        try{
          const p = document.querySelector(".p");
          p.style.color="red";
          //捕获错误 但不会终止程序的执行
        }catch(e){
          console.log(e.message); // e 根据浏览器提供的错误信息进行提示
          //return //使用return 终止程序
          throw new Error(e.message); //也可以搭配throw使用
        }
        finally{
          //一定会执行 ，不论对错
          alert("aaa");
        }
        console.log("123"); 
      }
      fn();
      console.log("ababa");
    </script>
  </body>
</html>
